### ISSUE TITLE
The `unit-case` rule rewrites a unit whose case mapping is not one character for one, changing the unit rather than its case

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#653]
### What minimal example or steps are needed to reproduce the bug?

A unit whose upper case is spelled with more characters than its lower case:

```css
a { width: 10Aß; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/unit-case": "upper"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

Nothing reported. `Aß` has no upper case that is still `Aß` — `"Aß".toUpperCase()` is `"ASS"`, three characters for two — so the rule has no recasing of that unit to ask for, only a different unit to write.

### What actually happened?

```
1:15-18  Expected "Aß" to be "ASS"
```

and `--fix` writes `a { width: 10ASS; }`. `Aß` and `ASS` are two different units: `lightningcss` prints each as it stands, `10Aß` and `10ASS`, and so does `less@4.9.1`. The declaration means something else after the fix.

The same holds the other way with a code point whose lower case is longer: under `lower`, `a { width: 10İ; }` is reported as `Expected "İ" to be "i̇"` and `--fix` writes `10i̇`, the dotted capital I becoming an `i` and a combining dot above — two code points where the file had one.

### What is the cause?

The rule asks for `unit.toLowerCase()` or `unit.toUpperCase()` and compares that with the unit, so any code point whose case mapping is not one character for one is reported and written. Neither mapping is the identifier's: a CSS unit is matched case-insensitively by the simple case folding of its code points, and `ß` folds to itself.

Measured on `main` at `c285324`, under `@stylistic/unit-case` and under the `scss`, `less` and `styled` namespaces alike, in a declaration and in a set of media parameters.

### Do you have a proposal to fix the bug?

Report a unit only where its recased form is the same run of characters in another case — that is, where the mapping is one code point for one. What that costs is worth measuring first: the units of CSS are ASCII, so the narrowing should reach nothing a stylesheet really spells, and a sweep over the corpus of [#649](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/649) would say so.
